tensorboard/components/tf_storage/storage.ts

```diff
@@ -184,13 +213,12 @@ function makeBindings<T>(fromString: (string) => T, toString: (T) => string): {
       // if this is a localStorage property, we always synchronize the value
       // in localStorage to match the one currently in the URI.
       if (fullOptions.useLocalStorage) {
-        set(uriStorageName, newVal, true);
-      }
-      if (!_.isEqual(newVal, get(uriStorageName, false))) {
+        set(uriStorageName, newVal, fullOptions);
+      } else if (!_.isEqual(newVal, get(uriStorageName, {useLocalStorage: false}))) {
         if (_.isEqual(newVal, fullOptions.defaultValue)) {
```

Comment:
Here we unset the URI key if we're setting the value to the default value, but we only do that if the new value to set wasn't the same as the existing URI value.  So in a case where the URI does contain the default value as an explicit parameter (e.g. the user manually inputs it), we won't clear it from the URI here (or in the hashchange listener).  Maybe we should be doing that?

Reply:
This behavior felt super super weird to me. Do you see a strong value in it? The application, without the hash whose value will equal to the defaultValue, will behave correctly.